### PR TITLE
feat: [PCFG-301] Check ICA

### PR DESCRIPTION
**Pull request by aomegax**
#### List of Changes
- added backend functionalities about check ICA

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/pagopa/apiconfig/model/creditorinstitution/IcaXml.java [hunk 1]
```diff
@@ -1,40 +1,50 @@
...
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+import java.time.LocalDateTime;
 import java.util.List;
 
 @XmlRootElement(name = "informativaContoAccredito")
 @XmlAccessorType(XmlAccessType.FIELD)
-@AllArgsConstructor
-@NoArgsConstructor
-@Builder
+//@AllArgsConstructor
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/model/creditorinstitution/IcaXml.java:22]: remove commented code

File: src/test/java/it/pagopa/pagopa/apiconfig/service/IcaServiceTest.java [hunk 2]
```diff
@@ -158,4 +169,88 @@ void deleteIca() {
...
+        List<IbanValidiPerPa> ibans = List.of(iban_1, iban_2);
+
+        when(ibanValidiPerPaRepository.findAllByFkPa(anyLong())).thenReturn(ibans);
+
+        List<CheckItem> checkItemList = icaService.verifyIca(file);
+        assertEquals(3,checkItemList.stream().filter(item -> item.getValid().equals(CheckItem.Validity.NOT_VALID)).count());
+    }
+
+    //
+//    @Test
```
Comment (jacopocarlini) [inline src/test/java/it/pagopa/pagopa/apiconfig/service/IcaServiceTest.java:244]: why are some tests commented? Can we remove them?

File: src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java [hunk 3]
```diff
@@ -130,19 +141,17 @@ public XSDValidation verifyXSD(MultipartFile xml) {
 
     @Transactional
     public void createIca(@NotNull MultipartFile file) {
-        // syntactic checks
-        checkSyntax(file);
+        List<CheckItem> checks = verifyIca(file);
+
+        Optional<CheckItem> check = checks.stream().filter(item -> item.getValid().equals(CheckItem.Validity.NOT_VALID)).findFirst();
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java:146]: Just a hint: put each method in a differente line. In this way in a stacktrace it's easier to understand where is the problem.

example:
```suggestion
        Optional<CheckItem> check = checks.stream()
        .filter(item -> item.getValid().equals(CheckItem.Validity.NOT_VALID))
        .findFirst();
```

**Review by jacopocarlini — COMMENTED** [review]
Good job, very complex task

File: src/main/java/it/pagopa/pagopa/apiconfig/model/creditorinstitution/IcaXml.java (hunk 1 above)
Comment (aomegax) [reply src/main/java/it/pagopa/pagopa/apiconfig/model/creditorinstitution/IcaXml.java:22]: done

File: src/test/java/it/pagopa/pagopa/apiconfig/service/IcaServiceTest.java (hunk 2 above)
Comment (aomegax) [reply src/test/java/it/pagopa/pagopa/apiconfig/service/IcaServiceTest.java:244]: cut & paste 😄

File: src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java (hunk 3 above)
Comment (aomegax) [reply src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java:146]: ok 👍

**Pull request merged**